### PR TITLE
Moving XMLCodec constants to their own classes

### PR DESCRIPTION
**Pull request by sambish5**
https://github.com/NationalSecurityAgency/emissary/pull/515#discussion_r1276517975
Done from this comment left on 515. Moved AnswerFile constants and method constants to separate classes.

### CODE REVIEW

**Comment by drivenflywheel** [comment]
> Some suggestions...
> 
> * "AsnwerXMLFileElementNames" -> "IbdoXmlElementNames"
Disagree on this one.  IBDOs don't have element names, and these names are specific to our UnitTest/ExtractionTest/RegressionTest XML answer files.
> * "IBDOMethodNames" -> "IbdoMethodNames"
Meh, I find IBDOMethodNames to be more readable
> * Javadoc all of the constants in the above classes.
IMO, constants should only have Javadoc comments if those comments clarify usage or ambiguity.
> * Make the above two classes final.
> * Move ...ATTRIBUTE_NAME fields to IbdoXmlElementNames class.

**Review by drivenflywheel — CHANGES_REQUESTED** [review]
Good first pass, but I think we can further improve the readability with minimal effort.

File: src/main/java/emissary/core/constants/AnswerXMLFileElementNames.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.AbstractMap;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class AnswerXMLFileElementNames {
+    // common XML element names
+    public static final String ANSWERS_ELEMENT_NAME = "answers";
```
[the remark this answers is not in the document]
Comment (sambish5) [reply src/main/java/emissary/core/constants/AnswerXMLFileElementNames.java:14]: Would you think it'd be smart to do this in IbdoMethodNames as well, to remove the `_METHOD_NAME` part? I think I'd keep the `SET` to keep it obvious they are setters

[the remark this answers is not in the document]
Comment (drivenflywheel) [reply src/main/java/emissary/core/constants/AnswerXMLFileElementNames.java:14]: Yes, absolutely.  I thought my last review included a specific comment that we drop those suffixes and move `SET_` from the end to the front to mirror the actual method names. Only one referencing file to update. Good catch!

**Pull request merged**